pkg/manager-nvme/manager.go

```diff
@@ -191,6 +195,29 @@ func findStoragePool(storageId, storagePoolId string) (*Storage, *interface{}) {
 	return nil, nil
 }
 
+func CleanupVolumes(volumesToKeep []PersistentVolumeInfo) error {
+	// For each storage device, build a list of namespaces to keep and then remove everything else
+	for _, storage := range mgr.storage {
+		if !storage.IsEnabled() {
+			continue
+		}
+
+		var namespacesToKeep []nvme.NamespaceIdentifier
+		for _, keepVolume := range volumesToKeep {
+			if keepVolume.SerialNumber == storage.serialNumber {
+				namespacesToKeep = append(namespacesToKeep, keepVolume.NamespaceId)
+			}
+		}
+
+		if err := storage.CleanupNamespaces(namespacesToKeep); err != nil {
+			// TODO return error here or just move to the next storage device?
+			return fmt.Errorf("Cleanup Volumes: Failed to remove abandoned namespaces on storage %s: %v", storage.id, err)
```

Comment:
I'm learning toward just logging the error and moving on rather than bomb out with an error. Thoughts?

Reply:
Agreed - the failure seems highly unlikely